### PR TITLE
Update for SQL and PPL parser to accept `simple_query_string`

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
### Description
SQL and PPL parser update to support `simple_query_string`, see https://github.com/opensearch-project/sql/issues/192
Require #54 to support end-to-end flow.

TODO:
* tests
* rework on `AstExpressionBuilder.java` x2

### Issues Resolved
^
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ppl/src/main/antlr/OpenSearchPPLParser.g4 [hunk 1]
```diff
@@ -216,7 +216,10 @@ booleanExpression
 
 relevanceExpression
     : relevanceFunctionName LT_PRTHS
-        field=relevanceArgValue COMMA query=relevanceArgValue
+        field=relevanceField COMMA query=relevanceQuery
+        (COMMA relevanceArg)* RT_PRTHS
+    | relevanceFunctionNameEx LT_PRTHS
```
Comment (Yury-Fridlyand) [inline ppl/src/main/antlr/OpenSearchPPLParser.g4:221]: Should we rename it? Like `relevanceFunctionType1Name` and `relevanceFunctionType2Name`.
Or maybe to make `relevanceExpressionType1` and `relevanceExpressionType2`.

File: ppl/src/main/java/org/opensearch/sql/ppl/parser/AstExpressionBuilder.java [hunk 2]
```diff
@@ -332,8 +335,11 @@ private List<UnresolvedExpression> relevanceArguments(RelevanceExpressionContext
...
     ImmutableList.Builder<UnresolvedExpression> builder = ImmutableList.builder();
+    ParserRuleContext field = ctx.field;
+    if (field == null)
+      field = ctx.fields;
     builder.add(new UnresolvedArgument("field",
-        new Literal(StringUtils.unquoteText(ctx.field.getText()), DataType.STRING)));
+        new Literal(StringUtils.unquoteText(field.getText()), DataType.STRING)));
     builder.add(new UnresolvedArgument("query",
         new Literal(StringUtils.unquoteText(ctx.query.getText()), DataType.STRING)));
     ctx.relevanceArg().forEach(v -> builder.add(new UnresolvedArgument(
```
Comment (Yury-Fridlyand) [inline ppl/src/main/java/org/opensearch/sql/ppl/parser/AstExpressionBuilder.java:345]: 1. Should we create 2 different functions as `relevanceArguments`?
2. Should we skip `StringUtils.unquoteText` for `fields`?

**Comment by MaxKsyunz** [comment]
Could you please make the SQL Java CI actions pass?

File: ppl/src/main/antlr/OpenSearchPPLParser.g4 (hunk 1 above)
Comment (acarbonetto) [reply ppl/src/main/antlr/OpenSearchPPLParser.g4:221]: Please be explicit with your naming as possible.  Type1 and Type2 don't mean anything.  If there's an order associated with the types in the list (and no other way to differentiate the types) we can call them first and second or something similar.

File: ppl/src/main/java/org/opensearch/sql/ppl/parser/AstExpressionBuilder.java [hunk 3]
```diff
@@ -332,8 +336,12 @@ private List<UnresolvedExpression> relevanceArguments(RelevanceExpressionContext
     // all the arguments are defaulted to string values
     // to skip environment resolving and function signature resolving
     ImmutableList.Builder<UnresolvedExpression> builder = ImmutableList.builder();
+    ParserRuleContext field = ctx.field;
+    if (field == null) {
```
Comment (acarbonetto) [inline ppl/src/main/java/org/opensearch/sql/ppl/parser/AstExpressionBuilder.java:340]: use conditions to define the ParseRuleContext, for example: 
ParserRuleContext field = ctx.field != null ? ctx.field : ctx.fields;

File: sql/src/main/antlr/OpenSearchSQLLexer.g4 [hunk 4]
```diff
@@ -357,6 +358,8 @@ BIT_XOR_OP:                         '^';
 DOT:                                '.';
 LR_BRACKET:                         '(';
 RR_BRACKET:                         ')';
+LT_SQR_PRTHS:                       '[';
```
Comment (acarbonetto) [inline sql/src/main/antlr/OpenSearchSQLLexer.g4:361]: where's this name coming from?
Just wondering why not something like: `LR_SQUARE_BRACKET` and `RR_SQUARE_BRACKET`

File: sql/src/main/java/org/opensearch/sql/sql/parser/AstExpressionBuilder.java [hunk 5]
```diff
@@ -362,8 +364,12 @@ public UnresolvedExpression visitConvertedDataType(
 
   @Override
   public UnresolvedExpression visitRelevanceFunction(RelevanceFunctionContext ctx) {
+    ParserRuleContext func = ctx.relevanceFunctionNameEx();
```
Comment (acarbonetto) [inline sql/src/main/java/org/opensearch/sql/sql/parser/AstExpressionBuilder.java:367]: `ParserRuleContext func = ctx.relevanceFunctionNameEx() != null ctx.relevanceFunctionNameEx() : ctx.relevanceFunctionName();

File: sql/src/main/java/org/opensearch/sql/sql/parser/AstExpressionBuilder.java [hunk 6]
```diff
@@ -393,8 +399,12 @@ private List<UnresolvedExpression> relevanceArguments(RelevanceFunctionContext c
     // all the arguments are defaulted to string values
     // to skip environment resolving and function signature resolving
     ImmutableList.Builder<UnresolvedExpression> builder = ImmutableList.builder();
+    ParserRuleContext field = ctx.field;
```
Comment (acarbonetto) [inline sql/src/main/java/org/opensearch/sql/sql/parser/AstExpressionBuilder.java:402]: ParserRuleContext field = ctx.field != null ctx.field : ctx.field2;

File: ppl/src/main/antlr/OpenSearchPPLParser.g4 (hunk 1 above)
Comment (MaxKsyunz) [reply ppl/src/main/antlr/OpenSearchPPLParser.g4:221]: Let's change it to have separate rule for single-field relevance functions and a separate rule for multi-field relevance functions.

This way we can avoid the cumbersome `?:` in `AstExpressionBuilder`. Instead there will be one method to visit single field relevance functions and another to visit multi-field relevance functions.

**Review by MaxKsyunz — CHANGES_REQUESTED** [review]

**Comment by Yury-Fridlyand** [comment]
Closed as of #61.

**Pull request closed without merging**